plugins/woocommerce/includes/class-wc-ajax.php

```diff
@@ -1255,7 +1255,11 @@ public static function remove_order_coupon() {
 				throw new Exception( __( 'Invalid coupon', 'woocommerce' ) );
 			}
 
-			$order->remove_coupon( wc_format_coupon_code( wp_unslash( $coupon ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			$code = wc_format_coupon_code( wp_unslash( $coupon ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			if ( $order->remove_coupon( $code ) ) {
+				// translators: %s coupon code.
+				$order->add_order_note( sprintf( __( 'Coupon removed: "%s".', 'woocommerce' ), $code ), 0, true );
```

Comment:
I don't think there's a huge risk here, but it might be slightly safer to do:

```suggestion
				$order->add_order_note( esc_html( sprintf( __( 'Coupon removed: "%s".', 'woocommerce' ), $code ), 0, true ) );
```